### ISSUE TITLE
 Question about cross-entropy

### ISSUE THREAD

**Issue by ZOUKaifeng** [probtorch/probtorch#49]
I read your article entitled "Learning Disentangled Representations with Semi-Supervised Deep Generative Models", and
I think that he could be of great interest for my research. I would like to adapt it to my own problem but I started to
test it on the MNIST database, as you did.

I first used my own implementation for the MNIST case (for simplicity and understanding) and disantlanglement was not working properly.
By checking your implementation, I observed that you use a binary cross entropy loss function for "x". However, as stated in your article,
this is a supposed to be a Gaussian continuous variable. Did I miss something?

Then,  I changed with my own implementation: in the loss function, I replace the Gaussian distribution for "x"  with the binary
cross entropy, then it makes it works!! 

Why disantlanglement requires a binary cross entropy loss function for "x" . It shoud be a Gaussian pdf. It makes no sense to me.
Do I miss something? Thanks a lot for your help.

**Comment by jwvdm** [comment]
This is indeed a point that is not very well explained in our paper (or indeed in the literature at large). 

Basically, when using a binary cross entropy, we "pretend" that the data is sampled from a Bernoulli likelihood, which equates to a cross-entropy loss. The mathematical interpretation of "pretend" here is a little subtle. The idea is that we can first assume a step that binarizes the input:
```
q(x' | x) = Bernoulli(x' ; x) 
```
In a standard VAE we would then evaluate the loss
```
E_{q(x' | x) q(z | x)}[ log p(x' | z) ] + KL(q(z | x) || p(z))
```
It turns out that the reconstruction loss in this objective is just the binary cross-entropy between the (non-binarized) x and the reconstruction. We do something analogous here in the semi-supervised case. 

You should be able to these models to work with either a Gaussian or a Bernoulli likelihood. Just remember that in a Gaussian likelihood you do have an extra degree of freedom, which is the noise σ in the likelihood, which effectively controls the strength of the reconstruction loss relative to the other terms in the objective.
